src/models/room.ts

```diff
@@ -1679,16 +1674,16 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         return this.setUnreadNotificationCount(type, count);
     }
 
-    public setSummary(summary: IRoomSummary | IRoomSummaryMSC4186): void {
+    /**
+     * Takes either a legacy or MSC4186 room summary and updates the room with it.
+     * The summary will be converted to a legacy style summary in order to be emitted
+     * in the `RoomEvent.Summary` event.
```

Comment:
this buries the lede

```suggestion
     * Note that the provided `summary` will be modified. (Specifically, it will be converted to a legacy-style 
     * `IRoomSummary`).
```